receiver/apachedruidreceiver/metrics.go

```diff
@@ -0,0 +1,306 @@
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	queryCountStats := make(map[string]float64)
+	for _, dataPoint := range metrics {
+		queryCountStats[dataPoint.Metric] = dataPoint.Value
+	}
+
+	mReceiver.metricsBuilder.RecordApachedruidQueryCountDataPoint(now, int64(queryCountStats[druidQueryCount]))
+	mReceiver.metricsBuilder.RecordApachedruidSuccessQueryCountDataPoint(now, int64(queryCountStats[druidQuerySuccessCount]))
+	mReceiver.metricsBuilder.RecordApachedruidFailedQueryCountDataPoint(now, int64(queryCountStats[druidQueryFailedCount]))
+	mReceiver.metricsBuilder.RecordApachedruidInterruptedQueryCountDataPoint(now, int64(queryCountStats[druidQueryInterruptedCount]))
+	mReceiver.metricsBuilder.RecordApachedruidTimeoutQueryCountDataPoint(now, int64(queryCountStats[druidQueryTimeoutCount]))
+	mReceiver.recordSQLQueryDataPoints(now, metrics)
+
+	return mReceiver.metricsBuilder.Emit(metadata.WithApachedruidService(service))
+}
+
+// remove all metrics published to the receiver besides the ones it cares about
+func purgeMetrics(metrics []datapoint) ([]datapoint, string) {
+	collectedMetrics := [7]string{druidQueryCount, druidQuerySuccessCount, druidQueryFailedCount, druidQueryInterruptedCount, druidQueryTimeoutCount, druidSQLQueryTime, druidSQLQueryBytes}
+	purgedArray := make([]datapoint, 0)
+	var service string
+
+	for _, dataPoint := range metrics {
+		if dataPoint.Metric == "" || !contains(collectedMetrics[:], dataPoint.Metric) {
+			continue
+		}
+
+		purgedArray = append(purgedArray, dataPoint)
+		if service == "" {
+			service = dataPoint.Service
+		}
+	}
+
+	return purgedArray, service
+}
+
+func (mReceiver *metricsReceiver) recordSQLQueryDataPoints(now pcommon.Timestamp, metrics []datapoint) {
+	dataSources := make([]string, 0)
+	sqlQueryCount := make(map[string]float64)
+	sqlQueryTime := make(map[string]float64)
+	sqlQueryBytes := make(map[string]float64)
+	for _, dataPoint := range metrics {
+		if dataSource, ok := dataPoint.DataSource.(string); ok {
+			if dataSource != "" {
+				switch dataPoint.Metric {
+				case druidSQLQueryTime:
+					sqlQueryTime[dataSource] += dataPoint.Value
+					sqlQueryCount[dataSource]++
+					if !contains(dataSources, dataSource) {
+						dataSources = append(dataSources, dataSource)
+					}
+				case druidSQLQueryBytes:
+					sqlQueryBytes[dataSource] += dataPoint.Value
+					if !contains(dataSources, dataSource) {
+						dataSources = append(dataSources, dataSource)
+					}
+				}
+			}
+		}
+	}
+
+	for _, source := range dataSources {
+		count := sqlQueryCount[source]
+		mReceiver.metricsBuilder.RecordApachedruidAverageSQLQueryTimeDataPoint(now, sqlQueryTime[source]/count, source)
+		mReceiver.metricsBuilder.RecordApachedruidAverageSQLQueryBytesDataPoint(now, sqlQueryBytes[source]/count, source)
+		mReceiver.metricsBuilder.RecordApachedruidSQLQueryCountDataPoint(now, int64(count), source)
+	}
+}
+
+func contains(s []string, e string) bool {
```

Comment:
It looks like all of your uses of contains could be replaced by initializing a map instead of a slice, and then just checking if the desired key exists in the map 
```
if _, ok := yourmap[key]; ok {

}
```